pages/view/_iscnId.vue

```diff
@@ -227,5 +350,86 @@ export default class ViewIscnIdPage extends Vue {
       this.$nuxt.error({ statusCode: 404, message: ErrorMessage.statusCode404 })
     }
   }
+
+  get record() {
+    return this.getISCNById(this.iscnId)?.data
+  }
+
+  get metadata() {
+    return this.record && (this.record as any).contentMetadata
+  }
+
+  get type() {
+    return this.metadata && this.metadata['@type']
+  }
+
+  get imgSrc() {
+    return (
+      (this.type === 'Image' || this.type === 'Photo') &&
+      getIPFSUrlFromISCN(this.getISCNById(this.iscnId))
+    )
+  }
+
+  get name() {
+    return this.metadata.name || this.metadata.title
+  }
+
+  get keywords(): Array<string> {
+    return this.metadata.keywords ? this.metadata.keywords.split(',') : []
+  }
+
+  get stakeholders() {
+    return this.record.stakeholders
+  }
+
+  get likerIdURL() {
+    return `${LIKER_LAND_URL}${this.stakeholderInfo.likerId}`
+  }
+
+  get transactionsURL() {
+    return `${BIG_DIPPER_TRANSACTIONS}${this.txHash}`
+  }
+
+  get rawDataURL() {
+    return `${ISCN_TX_RAW_DATA_ENDPOINT}${this.iscnId}`
+  }
+
+  async fetchTxHash() {
+    const txHash = await this.$axios.get(`${ISCN_TX_RAW_DATA_ENDPOINT}${this.iscnId}`)
+      .then((res) =>this.getTxHash(res.data.txs))
+    return txHash
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  getTxHash(item: any) {
+    let txhash = ''
+    item.forEach((element: any, i: any) => {
+      element.logs.forEach((logs: any) => {
+        logs.events.forEach((events: any) => {
+          events.attributes.forEach((attributes: any) => {
+            if (
+              attributes.key === 'action' &&
+              attributes.value === 'create_iscn_record'
+            ) {
+              txhash = item[i].txhash
+            }
+          })
+        })
+      })
+    })
+    return txhash
+  }
```

Comment:
Should use a simple `for loop` + `.find` for early exit, `forEach` forces you to loop through everything